packages/studio/src/components/ErrorComponentPreview.tsx

```diff
@@ -1,32 +1,63 @@
-import { ErrorComponentState } from "@yext/studio-plugin";
-import { useCallback, useMemo, useRef, useState } from "react";
+import {
+  useCallback,
+  useMemo,
+  useRef,
+  useState,
+  useEffect,
+  Dispatch,
+  SetStateAction,
+} from "react";
 import { v4 } from "uuid";
 import ErrorBoundary from "./common/ErrorBoundary";
-import { Tooltip } from "react-tooltip";
+import { ErrorComponentState } from "@yext/studio-plugin";
+import { ITooltip } from "react-tooltip";
 
 export default function ErrorComponentPreview(props: {
   errorComponentState: ErrorComponentState;
   element: JSX.Element | null;
+  setTooltipProps: Dispatch<SetStateAction<ITooltip>>;
 }) {
-  const { errorComponentState, element } = props;
+  const { errorComponentState, element, setTooltipProps } = props;
   const containerRef = useRef<HTMLDivElement>(null);
   const uniqueId = useMemo(() => v4(), []);
   // We cannot use the uuid on ErrorComponentState due to
   // repeaters re-using the same uuid
   const anchorId = `ErrorComponentState-${uniqueId}`;
   const [tooltipOpen, setTooltipOpen] = useState(false);
+  const closeTooltip = () => setTooltipOpen(false);
   const onMouseEnter = useCallback(() => {
     setTooltipOpen(true);
   }, []);
-  const onMouseLeave = useCallback(() => {
-    setTooltipOpen(false);
+  const onMouseLeave = useCallback(closeTooltip, []);
+
+  useEffect(() => {
+    document
+      .querySelector("iframe")
+      ?.contentDocument?.addEventListener("scroll", closeTooltip);
+    document.addEventListener("scroll", closeTooltip);
+    return () => {
+      document
+        .querySelector("iframe")
+        ?.contentDocument?.removeEventListener("scroll", closeTooltip);
+      document.removeEventListener("scroll", closeTooltip);
+    };
   }, []);
 
-  const rect = containerRef.current?.getBoundingClientRect();
-  const tooltipPosition = rect && {
-    x: (rect.right + rect.left) / 2,
-    y: rect.top - 40,
-  };
+  useEffect(() => {
+    const rect = containerRef.current?.getBoundingClientRect();
+    const y = (rect && rect.top + 25 - window.scrollY) || 0;
+    const tooltipPosition = rect && {
+      x: (rect.right + rect.left) / 2 + window.innerWidth / 4 - window.scrollX,
+      y: y < 25 ? 25 : y,
+    };
+
+    setTooltipProps({
+      isOpen: tooltipOpen,
+      content: errorComponentState.message,
+      anchorSelect: anchorId,
+      position: tooltipPosition || { x: 0, y: 0 },
```

Comment:
nit: can we use nullish coalescing operator (`??`) here instead of logical OR? that makes it a little more clear to me when the default value should be used